pyteal/ast/router.py

```diff
@@ -297,12 +297,11 @@ def to_cond_node(self, use_frame_pt: bool = False) -> CondNode:
 
 class ASTBuilder:
     def __init__(self):
-        self.conditions_n_branches: list[CondNode] = []
```

Comment:
`self.conditions_n_branches` are used (updated) only in `program_construction`, so we only keep things that leads to computation result, rather than intermediate artifact.